repo/plugin.video.nzbdav/resources/lib/webdav.py

```
@@ -693,14 +802,47 @@ def find_video_stream_for_folder(
+        )
+        if requested_episode is not None:
+            # Explicit route identity is authoritative. A named different
+            # episode, or an incomplete scan, must never degrade to the
+            # title-derived/legacy candidate from find_video_file().
+            video_path = inventory.selected_path if inventory is not None else None
+            if inventory is not None and video_path:
+                _remember_video_file_size_hint(video_path, inventory.selected_size)
+    if on_inventory is not None and inventory is not None:
+        try:
+            on_inventory(inventory)
+        except Exception as error:  # pylint: disable=broad-except
```

Ruff (0.15.20)

[warning] Do not catch blind exception: `Exception`

(BLE001)